deepchecks/base/check.py

```diff
@@ -85,41 +87,83 @@ def __init__(self, value, header: str = None, display: Any = None):
+        """Display the check result or return the display as widget.
+
+        Args:
+            unique_id (str):
+                The unique id given by the suite that displays the check.
+            as_widget (bool):
+                Boolean that controls if to display the check regulary or if to return a widget.
+            show_additional_outputs (bool):
+                Boolean that controls if to show additional outputs.
+        Returns:
+            Widget representation of the display if as_widget is True.
+        """
+        if as_widget:
+            box = widgets.VBox()
+            box_children = []
+        check_html = ''
         if unique_id:
             check_id = f'{self.check.__class__.__name__}_{unique_id}'
-            display_html(f'<h4 id="{check_id}">{self.get_header()}</h4>', raw=True)
+            check_html += f'<h4 id="{check_id}">{self.get_header()}</h4>'
         else:
-            display_html(f'<h4>{self.get_header()}</h4>', raw=True)
+            check_html += f'<h4>{self.get_header()}</h4>'
         if hasattr(self.check.__class__, '__doc__'):
-            docs = self.check.__class__.__doc__ or ''
-            # Take first non-whitespace line.
-            summary = next((s for s in docs.split('\n') if not re.match('^\\s*$', s)), '')
-            display_html(f'<p>{summary}</p>', raw=True)
-        if self.conditions_results and show_conditions:
-            display_html('<h5>Conditions Summary</h5>', raw=True)
-            display_conditions_table(self, unique_id)
+            summary = get_docs_summary(self.check)
+            check_html += f'<p>{summary}</p>'
+        if self.conditions_results:
+            check_html += '<h5>Conditions Summary</h5>'
+            check_html += get_conditions_table_display(self, unique_id)
+            check_html += '<h5>Additional Outputs</h5>'
         if show_additional_outputs:
-            display_html('<h5>Additional Outputs</h5>', raw=True)
             for item in self.display:
                 if isinstance(item, (pd.DataFrame, Styler)):
-                    display_dataframe(item)
+                    check_html += dataframe_to_html(item)
                 elif isinstance(item, str):
-                    display_html(item, raw=True)
+                    check_html += item
                 elif isinstance(item, BaseFigure):
-                    item.show()
+                    if as_widget:
+                        box_children.append(widgets.HTML(check_html))
+                        box_children.append(go.FigureWidget(data=item))
+                    else:
+                        display_html(check_html, raw=True)
+                        item.show()
+                    check_html = ''
                 elif callable(item):
                     try:
                         item()
-                        plt.show()
+                        if as_widget:
+                            plt_out = widgets.Output()
```

Comment:
by the way, did you test it? I remember I tried something similar and couldn't make it work